utils/rand/rand.go

```diff
@@ -0,0 +1,169 @@
+package rand
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"errors"
+	"fmt"
+)
+
+// This package is a wrppaer around true RNG crypto/rand.
+// It implements useful tools using the true RNG and that
+// are not exported by the crypto/rand package.
+// This package does not implement any determinstic RNG (Pseudo RNG)
+// unlike the package flow-go/crypto/random.
+
+var randFailure = errors.New("crypto/rand failed")
+
+// returns a random uint64
+func Uint64() (uint64, error) {
+	buffer := make([]byte, 8) // TODO: declare as a global variable and add a lock?
+	if _, err := rand.Read(buffer); err != nil {
+		return 0, randFailure
+	}
+	r := binary.LittleEndian.Uint64(buffer)
+	return r, nil
+}
+
+// returns a random uint64 strictly less than n
+// errors if n==0
+func Uint64n(n uint64) (uint64, error) {
+	if n == 0 {
+		return 0, fmt.Errorf("n should be strictly positive, got %d", n)
+	}
+	// the max returned random is n-1 > 0
+	max := n - 1
+	// count the bytes size of max
+	size := 0
+	for tmp := max; tmp != 0; tmp >>= 8 {
+		size++
+	}
+	buffer := make([]byte, 8) // TODO: declare as a global variable and add a lock?
```

Comment:
if you change to `buffer := make([]byte, size)`, you can skip doing `buffer[:size]` at line 58